packages/devtools-launchpad/src/components/LandingPage.js

```diff
@@ -45,32 +45,59 @@ const LandingPage = React.createClass({
     };
   },
 
+  componentDidUpdate() {
+    this.refs.filterInput.focus();
+  },
+
   onFilterChange(newFilterString) {
     this.props.onFilterChange(newFilterString);
   },
 
   onSideBarItemClick(itemTitle) {
-    this.setState({ selectedPane: itemTitle });
-    this.onFilterChange("");
+    if (itemTitle !== this.state.selectedPane) {
+      this.setState({ selectedPane: itemTitle });
+      this.onFilterChange("");
+    }
+  },
+
+  onTabClick(tab, paramName) {
+    this.props.onTabClick(getTabURL(tab, paramName));
   },
 
   renderTabs(tabs, paramName) {
     if (!tabs || tabs.count() == 0) {
       return dom.div({}, "");
     }
 
+    let tabClassNames = ["tab"];
+    if (tabs.size === 1) {
+      tabClassNames.push("active");
+    }
+
     return dom.div(
       { className: "tab-group" },
       dom.ul(
         { className: "tab-list" },
-        tabs.valueSeq().map(tab => dom.li(
-          { "className": "tab",
-            "key": tab.get("id"),
-            "onClick": () => this.props.onTabClick(getTabURL(tab, paramName))
+        tabs.valueSeq().map(
+          tab => dom.li({
```

Comment:
The first render is handled by the `autoFocus` attribute on the input. I thought `componentDidMount` was only fired after the first render. also, this wouldn't pick the "update" of the state we do when we change of panel.